src/main/java/teammates/ui/webapi/output/StudentData.java

```diff
@@ -116,12 +116,29 @@ public void hideLastName() {
         setLastName(null);
     }
 
+    /**
+     * Hides some attributes to instructor on a search.
+     */
+    public void hideInformationForInstructorSearch() {
+        hideInformationForInstructor();
+        setComments(null);
+        setInstitute(null);
+    }
+
     /**
      * Hides some attributes to instructor.
      */
     public void hideInformationForInstructor() {
         setGoogleId(null);
+        setKey(null);
```

Comment:
I don't think this is necessary because the registration key is not set as part of constructor (i.e. it needs to be explicitly set).
But having said that, `googleId` also needs to be treated the same, i.e. needs to be explicitly set. This will also make the line before this redundant.